vital/types/metadata.cxx

```diff
@@ -56,18 +56,17 @@ namespace vital {
       : metadata_item( "Requested metadata item is not in collection", 0, VITAL_META_UNKNOWN )
     { }
 
-    virtual bool is_valid() const { return false; }
-    virtual vital_metadata_tag tag() const { return static_cast< vital_metadata_tag >(0); }
-    virtual std::type_info const& type() const { return typeid( void ); }
-    virtual std::string as_string() const { return "--Unknown metadata item--"; }
-    virtual double as_double() const { return 0; }
-    virtual double as_uint64() const { return 0; }
-    virtual std::ostream& print_value(std::ostream& os) const
+    bool is_valid() const override { return false; }
+    std::type_info const& type() const override { return typeid( void ); }
+    std::string as_string() const override { return "--Unknown metadata item--"; }
+    std::ostream& print_value(std::ostream& os) const override
     {
       os << this->as_string();
       return os;
     }
 
+    metadata_item* clone() const override { return const_cast< unknown_metadata_item* >( this ); }
```

Comment:
I believe that this could cause problems in Python. If a Python user, for some reason, wants to add an unknown_metadata_item to a metadata collection, there will exist 2 separate smart pointers pointing to the same object (one from the Python side managing the original object, one in the metadata collection). This could cause a double free. Would it be best to return a copy, like in the other clone() implementations? I admit that this is not how the class is meant to be used, but it could be beneficial to have that extra precaution

Reply:
The is only a single instance of the unknown_metadata_item which is returned when the requested item is not present. This class should not be used beyond this case. But to your point, The clone() method was needed to satisfy the interface, and since it is never called, that can be made more obvious by returning a nullptr and adding a comment.

Reply:
Yup, and doing otherwise is dangerous (see the comment I typed before this discussion started). Returning `nullptr` definitely seems safer.

Reply:
I'm going to resolve this conversation as Matthew's comment below continues the discussion